packages/boot/test/bootstrapTests/benchmark-vaults.js

```diff
@@ -0,0 +1,45 @@
+// This is a version of bench-vaults-performance.js designed to run as a
```

Comment:
Moving `benchmarkerator.js` to a `tools` directory is probably right, but I think the open question is: the `tools` directory of which package?  On the other hand, I feel pretty strongly that `benchmark-vaults.js` should not go into a `tools` directory, since that's not what it is -- it's a specific exercise of specific functionality, much more like a test in that regard (note that `yarn bench` looks in the `test` directory for benchmarks to run and has for a while).  If we produce a lot more of these benchmarks (which is sort of the idea behind making it easy to create them by providing support tooling) I think a case could be made for conventionally having a `benchmarks` directory parallel to `test`.

I think moving the bootstrap test driver stuff into `tools` makes sense, especially if we'll want to be using that stuff from other packages.  Deep imports from `test` definitely feel sketchy.  On the other hand, I think it's plausible that the longer term evolution of the benchmark tools eventually phases out the driver stuff here in favor of alternatives better fit to purpose, so if the only place they're actually used goes back to being just the bootstrap tests, then keeping them where they are makes sense.  On the other other hand, completely moving the benchmark stuff off the bootstrap test driver utils, if that ever happens, is certainly not going to happen right away, so having the support stuff in a place intended for common use at least until then makes sense.

Reply:
>  `tools` directory of which package?

The lowest point a which all its dependencies are satisfied. I think that's this one, unless you want to make a higher one or move some of the supporting tools her into a lower one. Even so I'd save that for another PR.

> benchmark-vaults.js should not go into a tools directory, since that's not what it is -- it's a specific exercise of specific functionality, much more like a test in that regard (note that yarn bench looks in the test directory for benchmarks to run and has for a while).

Each to change where `yarn bench` looks.

I think a test should be something that passes or fails in CI. Are these perf tests that pass or fail or a tool for collecting performance data? If the latter, I think that points squarely to `tools` and not `tests`.

> I think a case could be made for conventionally having a benchmarks directory parallel to test.

No objection. That would satisfy my requirement about what a "test" is.

> I think moving the bootstrap test driver stuff into tools makes sense…
+1 to that paragraph

Reply:
> Are these perf tests that pass or fail or a tool for collecting performance data? 

They're not pass/fail tests, so that does argue for their exclusion from the `test` directory.

However, they are not tools for collecting perf data generally but actual perf tests of specific things, in the same way that tests are actual correctness tests of specific things.  In other words, I don't think "find out how long opening a vault takes" and "find out long doing a liquidation takes" and "find out how long installing a new contract takes" are operations that belong in a directory called `tools`.  In contrast, the benchmark framework itself (Benchmarkerator and its brethren) I think does belong in a `tools` directory in the same way that, say, a module for setting up Ava belongs there.

Reply:
Based on the above conversation, the resolution approach I'm going to take is: land this PR with things located where they are now, then do a follow-on PR that does all the file relocation and renaming as discussed (i.e., tooling and support code in `tools`, the benchmarks themselves in a new `benchmarks` directory).  I suspect the benchmark tooling may eventually want to be in a different package (possibly even a new package of its own), but I think we can reasonably defer that until we have better visibility onto how and where it's being used.